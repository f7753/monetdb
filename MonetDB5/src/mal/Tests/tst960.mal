# the chunk mechanism

S:= new(:void,:str);
setSequenceBase(S,0:oid);
insert(S,nil,"hello");
insert(S,nil,"brave");
insert(S,nil,"new");
insert(S,nil,"world");
print(S);

L:= batstr.length(S);
print(L);
L:= batstr.bytes(S);
print(L);

B:= batstr.==(S,"hello");
print(B);

B:= batstr.match(S,"hello");
print(B);

U:= batstr.toUpper(S);
print(U);

l:= batstr.toLower(U);
print(l);

trim:= batstr.trim(S); 
print(trim);
ltrim:= batstr.ltrim(S); 
print(ltrim);
rtrim:= batstr.rtrim(S); 
print(rtrim);

srch:= batstr.search(S,S); 
print(srch);
plus:= batstr.+(S,S); 
print(plus);
