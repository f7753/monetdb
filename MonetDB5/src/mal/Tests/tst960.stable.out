stdout of test 'tst960` in directory 'src/mal` itself:


# 23:37:05 >  
# 23:37:05 >  Mtimeout -timeout 60 Mserver "--config=/ufs/mk/monet5/src/mal/Tests/All.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5/Linux/lib/MonetDB5:/ufs/mk/opensource/MonetDB/Linux/lib/MonetDB" --set "gdk_dbfarm=/ufs/mk/opensource/MonetDB/Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/opensource/MonetDB/Linux/var/MonetDB5/log" --set mapi_port=40158 --set sql_port=57521 --set monet_prompt= --trace "--config=/ufs/mk/monet5/src/mal/Tests/All.conf" --dbname=mTests_src_mal  tst960.mal </dev/null
# 23:37:05 >  

# Monet Database Server V4.7.1
# Copyright (c) 1993-2005, CWI. All rights reserved.
# Compiled for i686-redhat-linux-gnu/32bit; dynamically linked.
# config:/ufs/mk/monet5/src/mal/Tests/All.conf
# dbfarm:/ufs/mk/opensource/MonetDB/Linux/var/MonetDB5/dbfarm
# dbname:mTests_src_mal
# Visit http://monetdb.cwi.nl/ for further information.
## the chunk mechanism
#
#S:= new(:void,:str);
#setSequenceBase(S,0:oid);
#insert(S,nil,"hello");
#insert(S,nil,"brave");
#insert(S,nil,"new");
#insert(S,nil,"world");
#print(S);
#
#L:= batstr.length(S);
#print(L);
#L:= batstr.bytes(S);
#print(L);
#
#B:= batstr.==(S,"hello");
#print(B);
#
#B:= batstr.match(S,"hello");
#print(B);
#
#U:= batstr.toUpper(S);
#print(U);
#
#l:= batstr.toLower(U);
#print(l);
#
#trim:= batstr.trim(S); 
#print(trim);
#ltrim:= batstr.ltrim(S); 
#print(ltrim);
#rtrim:= batstr.rtrim(S); 
#print(rtrim);
#
#srch:= batstr.search(S,S); 
#print(srch);
#plus:= batstr.+(S,S); 
#print(plus);
#
function user.main():int;	# 0  (main:int)
    S := bbp.new(:void,:str);	# 1 CMDBATnew (S:bat[:void,:str])<-($2:void)($3:str)
    bat.setSequenceBase(S,0:oid);	# 2 BKCsetSequenceBase ($4:bat[:void,:str])<-(S:bat[:void,:str])($5:oid)
    bat.insert(S,nil,"hello");	# 3 BKCinsert_bun ($6:bat[:void,:str])<-(S:bat[:void,:str])($7:void)($8:str)
    bat.insert(S,nil,"brave");	# 4 BKCinsert_bun ($9:bat[:void,:str])<-(S:bat[:void,:str])($7:void)($10:str)
    bat.insert(S,nil,"new");	# 5 BKCinsert_bun ($11:bat[:void,:str])<-(S:bat[:void,:str])($7:void)($12:str)
    bat.insert(S,nil,"world");	# 6 BKCinsert_bun ($13:bat[:void,:str])<-(S:bat[:void,:str])($7:void)($14:str)
    io.print(S);	# 7 IOprint_val ($15:int)<-(S:bat[:void,:str])
    L := batstr.length(S);	# 8 STRbatLength (L:bat[:void,:int])<-(S:bat[:void,:str])
    io.print(L);	# 9 IOprint_val ($17:int)<-(L:bat[:void,:int])
    L := batstr.bytes(S);	# 10 STRbatBytes (L:bat[:void,:int])<-(S:bat[:void,:str])
    io.print(L);	# 11 IOprint_val ($18:int)<-(L:bat[:void,:int])
    B := batstr.==(S,"hello");	# 12 STRbatEqualCst (B:bat[:void,:bit])<-(S:bat[:void,:str])($8:str)
    io.print(B);	# 13 IOprint_val ($20:int)<-(B:bat[:void,:bit])
    B := batstr.match(S,"hello");	# 14 STRbatmatchCst (B:bat[:void,:bit])<-(S:bat[:void,:str])($8:str)
    io.print(B);	# 15 IOprint_val ($21:int)<-(B:bat[:void,:bit])
    U := batstr.toUpper(S);	# 16 STRbatUpper (U:bat[:void,:str])<-(S:bat[:void,:str])
    io.print(U);	# 17 IOprint_val ($23:int)<-(U:bat[:void,:str])
    l := batstr.toLower(U);	# 18 STRbatLower (l:bat[:void,:str])<-(U:bat[:void,:str])
    io.print(l);	# 19 IOprint_val ($25:int)<-(l:bat[:void,:str])
    trim := batstr.trim(S);	# 20 STRbatStrip (trim:bat[:void,:str])<-(S:bat[:void,:str])
    io.print(trim);	# 21 IOprint_val ($27:int)<-(trim:bat[:void,:str])
    ltrim := batstr.ltrim(S);	# 22 STRbatLtrim (ltrim:bat[:void,:str])<-(S:bat[:void,:str])
    io.print(ltrim);	# 23 IOprint_val ($29:int)<-(ltrim:bat[:void,:str])
    rtrim := batstr.rtrim(S);	# 24 STRbatRtrim (rtrim:bat[:void,:str])<-(S:bat[:void,:str])
    io.print(rtrim);	# 25 IOprint_val ($31:int)<-(rtrim:bat[:void,:str])
    io.print("Start searches");	# 26 IOprint_val ($32:int)<-($33:str)
    srch := batstr.search(S,"hello");	# 27 STRbatstrSearchcst (srch:bat[:void,:int])<-(S:bat[:void,:str])($8:str)
    io.print(srch);	# 28 IOprint_val ($35:int)<-(srch:bat[:void,:int])
    srch := batstr.search(S,S);	# 29 STRbatstrSearch (srch:bat[:void,:int])<-(S:bat[:void,:str])(S:bat[:void,:str])
    io.print(srch);	# 30 IOprint_val ($36:int)<-(srch:bat[:void,:int])
    plus := batstr.+(S,S);	# 31 STRbatConcat (plus:bat[:void,:str])<-(S:bat[:void,:str])(S:bat[:void,:str])
    io.print(plus);	# 32 IOprint_val ($38:int)<-(plus:bat[:void,:str])
end main;	# 33  
#-------------------------#
# h	tmp_13		  # name
# void	str		  # type
#-------------------------#
[ 0@0,	  "hello"	  ]
[ 1@0,	  "brave"	  ]
[ 2@0,	  "new"		  ]
[ 3@0,	  "world"	  ]
#-----------------#
# h	tmp_14	  # name
# void	int	  # type
#-----------------#
[ 0@0,	  5	  ]
[ 1@0,	  5	  ]
[ 2@0,	  3	  ]
[ 3@0,	  5	  ]
#-----------------#
# h	tmp_15	  # name
# void	int	  # type
#-----------------#
[ 0@0,	  5	  ]
[ 1@0,	  5	  ]
[ 2@0,	  3	  ]
[ 3@0,	  5	  ]
#-----------------#
# h	tmp_14	  # name
# void	bit	  # type
#-----------------#
[ 0@0,	  true	  ]
[ 1@0,	  false	  ]
[ 2@0,	  false	  ]
[ 3@0,	  false	  ]
#-----------------#
# h	tmp_16	  # name
# oid	bit	  # type
#-----------------#
[ 0@0,	  true	  ]
[ 1@0,	  false	  ]
[ 2@0,	  false	  ]
[ 3@0,	  false	  ]
#-------------------------#
# h	tmp_14		  # name
# void	str		  # type
#-------------------------#
[ 0@0,	  "HELLO"	  ]
[ 1@0,	  "BRAVE"	  ]
[ 2@0,	  "NEW"		  ]
[ 3@0,	  "WORLD"	  ]
#-------------------------#
# h	tmp_17		  # name
# void	str		  # type
#-------------------------#
[ 0@0,	  "hello"	  ]
[ 1@0,	  "brave"	  ]
[ 2@0,	  "new"		  ]
[ 3@0,	  "world"	  ]
#-------------------------#
# h	tmp_18		  # name
# void	str		  # type
#-------------------------#
[ 0@0,	  "hello"	  ]
[ 1@0,	  "brave"	  ]
[ 2@0,	  "new"		  ]
[ 3@0,	  "world"	  ]
#-------------------------#
# h	tmp_19		  # name
# void	str		  # type
#-------------------------#
[ 0@0,	  "hello"	  ]
[ 1@0,	  "brave"	  ]
[ 2@0,	  "new"		  ]
[ 3@0,	  "world"	  ]
#-------------------------#
# h	tmp_20		  # name
# void	str		  # type
#-------------------------#
[ 0@0,	  "hello"	  ]
[ 1@0,	  "brave"	  ]
[ 2@0,	  "new"		  ]
[ 3@0,	  "world"	  ]
[ "Start searches" ]
#-----------------#
# h	tmp_21	  # name
# void	int	  # type
#-----------------#
[ 0@0,	  0	  ]
[ 1@0,	  -1	  ]
[ 2@0,	  -1	  ]
[ 3@0,	  -1	  ]
#-----------------#
# h	tmp_22	  # name
# void	int	  # type
#-----------------#
[ 0@0,	  0	  ]
[ 1@0,	  0	  ]
[ 2@0,	  0	  ]
[ 3@0,	  0	  ]
#-------------------------#
# h	tmp_22		  # name
# void	str		  # type
#-------------------------#
[ 0@0,	  "hellohello"	  ]
[ 1@0,	  "bravebrave"	  ]
[ 2@0,	  "newnew"	  ]
[ 3@0,	  "worldworld"	  ]

# 23:37:05 >  
# 23:37:05 >  Done.
# 23:37:05 >  

